File: c++/0150-evaluate-reverse-polish-notation/main.cpp
#include <cstdlib>
#include <iostream>
#include <vector>

#include "solution.cpp"

using namespace std;

int main() {
    Solution solution;

    vector<string> tokens1 = {"2", "1", "+", "3", "*"};
    cout << "9: " << solution.evalRPN(tokens1) << endl;

    vector<string> tokens2 = {"4", "13", "5", "/", "+"};
    cout << "6: " << solution.evalRPN(tokens2) << endl;

    vector<string> tokens3 = {"10", "6", "9", "3", "+", "-11", "*", "/", "*", "17", "+", "5", "+"};
    cout << "22: " << solution.evalRPN(tokens3) << endl;

    return EXIT_SUCCESS;
}

File: java/0150-evaluate-reverse-polish-notation/Main.java
public class Main {
    public static void main(String[] args) {
        Solution solution = new Solution();

        System.out.println("9: " + solution.evalRPN(new String[] { "2", "1", "+", "3", "*" }));
        System.out.println("6: " + solution.evalRPN(new String[] { "4", "13", "5", "/", "+" }));
        System.out.println("22: " + solution.evalRPN(new String[] { "10", "6", "9", "3", "+", "-11", "*", "/", "*", "17", "+", "5", "+" }));
    }
}
